### PR TITLE
Feature/refine lists and check class

### LINKED ISSUE #228: check list improvements
It is not really convenient that this is an R6 object, ideally it should just be a tibble. But there are good reasons for being the data frame, in particular that while the check is running, the status is not known and an R6 object can auto-update itself.

Maybe we want:
- [ ] an `as_tibble()` method
- [ ] custom printing of some columns of that tibble
- [ ] more intuitive way to inspect a single check
- [ ] more intuitive way to inspect a check group

### PR DESCRIPTION
**Pull request by gaborcsardi**
Closes #228.

### CODE REVIEW

**Comment by gaborcsardi** [comment]
@maelle OK, I think this is done.

File: R/check-class.R [hunk 1]
```diff
@@ -247,3 +280,56 @@ first_line <- function(x) {
   l <- strsplit(x, "\n", fixed = TRUE)
   vapply(l, "[[", "", 1)
 }
+
+#' Retrieve the result of R-hub checks
+#'
+#' @param ids One of the following:
+#'  - A single R-hub check id. It may also be the beginning of the hash
+#'    of the id (i.e. the part after the last dash), if that check was
```
Comment (maelle) [inline R/check-class.R]: this part is unclear to me, if it is the beginning why is the part after the last dash?

**Comment by maelle** [comment]
:eyes:

File: R/list.R [hunk 2]
```diff
@@ -6,10 +6,36 @@
 #'   [validate_email()].
 #' @param package `NULL`, or a character scalar. Can be used to restrict
 #'   the search for a single package.
-#' @param howmany How many check groups (checks submitted simultaneously) to show. The current API limit is 20.
-#' @return An [`rhub_check_list`] object.
+#' @param howmany How many check groups (checks submitted simultaneously)
+#'   to show. The current API limit is 20.
+#' @return A tibble with columns:
```
Comment (maelle) [inline R/list.R]: should tibble here refer to tibble docs i.e. `[tibble::tibble]`? It'd look good in the pkgdown website at least.

File: R/check-class.R (hunk 1 above)
Comment (gaborcsardi) [reply R/check-class.R]: explained better in `?"rhub-ids"`

**Pull request merged**